xgcm/grid.py

```diff
@@ -352,43 +354,74 @@ def face_edge_data(fnum, face_axis, count=1):
             edge_data = concatenate(arrays, face_axis_num)
         else:
             edge_data = face_edge_data(None, None)
-
+        if self._periodic:
+            if boundary_discontinuity:
+                if is_left_edge:
+                    edge_data = edge_data - boundary_discontinuity
+                elif not is_left_edge:
+                    edge_data = edge_data + boundary_discontinuity
+
+        # Check consistency of output data
+        if not isinstance(edge_data, type(da.data)):
+            raise RuntimeError('Types do not match edge(%s)/da.data(%s)'
+                               % (type(edge_data), type(da.data)))
```

Comment:
How would we ever hit this situation? I'm not sure I can follow what is happening here.